### ISSUE TITLE
您好 损失不下降？

### ISSUE THREAD

**Issue by zwb170010** [MasLiang/Learning-without-Forgetting-using-Pytorch#2]
您好 损失为什么不下降？是否损失计算有问题？

**Comment by zwb170010** [comment]
您好 损失不下降？

**Comment by MasLiang** [comment]
Loss Function consists of two parts. The first part is the same as other models using CrossEntropy between output and target. 

Let's consider the equation of CrossEntropy.

$$L=\sum ylog(\hat y)$$

For $y$, only the true class is '1' and others are '0'. So there is only one term in this equation:

$$L=log(\hat y_{trueclass})$$

So the first part is small, and it is important to learn new knowledge. But this part will lead to great change of weight and bias.

Then consider the second part. The loss function is also CrossEntropy, but in this part, $y$ is not the ground truth, but output of the old model in which almost all values are not zero. So there are $n$ terms in this formula, and it will be huge. In this code, $n$ is 1000.
 
The second part is the distillation part which can help to hold the performance of the model on old classes. So we hope it not change.

That is to say we try to hold the larger part unchanged but change the smaller part. So it seems that the loss doesn't change.

**Comment by MasLiang** [comment]
I think may be you can inspect each part in loss function separately.

**Comment by MasLiang** [comment]
Maybe you can inspect each part of the loss function separately : )
